docs/contributing/coding-standards/component-options.md

```diff
@@ -4,204 +4,42 @@ Configuration options allow developers to pass additional information into a com
-
-See ['Naming configuration options'](#naming-configuration-options) for exceptions to how names are transformed.
-
-As we expect configuration-related `data-*` attributes to always be on the component's root element (the same element with the `data-module` attribute), we can access them all using `this.$root.dataset`.
-
-Using the `mergeConfigs` call discussed earlier in this document, update it to include `this.$root.dataset` as the highest priority after calling the constuctor of `GOVUKFrontendComponent` with `super()`.
-
-```mjs
-import { mergeConfigs, normaliseDataset } from '../../common/configuration.mjs'
-import { GOVUKFrontendComponent } from '../../govuk-frontend-component.mjs'
-
-export class Accordion extends GOVUKFrontendComponent{
-  constructor($root, config = {}) {
-    super($root)
-
-    this.config = mergeConfigs(
-      Accordion.defaults,
-      config,
-      normaliseDataset(Accordion, this.$root.dataset)
-    )
-  }
-}
-```
-
-Here, we pass the value of `this.$root.dataset` through our `normaliseDataset` function. This is because attribute values in dataset are always interpreted as strings. `normaliseDataset` looks at the component's configuration schema and converts values into numbers or booleans where needed.
-
-Now, in our HTML, we could pass configuration options by using the kebab-case version of the option's name.
-
-```html
-<div
-  class="govuk-accordion"
-  data-module="govuk-accordion"
-  data-remember-expanded="false">
-  ...
-</div>
-```
-
-However, this only works for developers who are writing raw HTML. We include Nunjucks macros for each component with GOV.UK Frontend to make development easier and faster, but this also makes it harder for developers to manually alter the raw HTML. We'll add a new parameter to Nunjucks to help them out.
-
-### Adding a configuration schema
-
-Components that accept configuration using `data-*` attributes also require a schema. This schema documents what parameters a configuration object may contain and what types of value they're expected to be.
-
-Having a schema is required for the `normaliseDataset` function to work. A schema is also needed to use the `validateConfig` and `extractConfigByNamespace` functions we'll cover later on.
-
-```mjs
-export class Accordion {
-  static schema = Object.freeze({
-    properties: {
-      i18n: { type: 'object' },
-      rememberExpanded: { type: 'boolean' }
-    }
-  })
-}
-```
-
-### Validating a provided configuration against the schema
+## Preparing the component's JavaScript
 
-You can use the `validateConfig` function to ensure that a configuration object matches the schema format.
+If a component class needs configuration, it [can extend the `ConfigurableComponent` class](https://frontend.design-system.service.gov.uk/building-configurable-components/) that GOV.UK Frontend provides. This class accepts configuration from the following sources:
```

Comment:
**nitpick** In the context of our internal configuration to describe how to create components for GOV.UK Frontend, I think extending `ConfigurableComponent` is more mandated than optional.

```suggestion
If a component class needs configuration, [extend the `ConfigurableComponent` class](https://frontend.design-system.service.gov.uk/building-configurable-components/) that GOV.UK Frontend provides. This class accepts configuration from the following sources:
```